0.1.1 won't build for me no mo'!

```
mix nerves.new dht_test
* creating dht_test/config/config.exs
* creating dht_test/config/target.exs
* creating dht_test/lib/dht_test.ex
* creating dht_test/lib/dht_test/application.ex
* creating dht_test/test/test_helper.exs
* creating dht_test/test/dht_test_test.exs
* creating dht_test/rel/vm.args.eex
* creating dht_test/rootfs_overlay/etc/iex.exs
* creating dht_test/.gitignore
* creating dht_test/.formatter.exs
* creating dht_test/mix.exs
* creating dht_test/README.md

Fetch and install dependencies? [Yn] Y
* running mix deps.get
Your Nerves project was created successfully.

You should now pick a target. See https://hexdocs.pm/nerves/targets.html#content
for supported targets. If your target is on the list, set `MIX_TARGET`
to its tag name:

For example, for the Raspberry Pi 3 you can either
  $ export MIX_TARGET=rpi3
Or prefix `mix` commands like the following:
  $ MIX_TARGET=rpi3 mix firmware

If you will be using a custom system, update the `mix.exs`
dependencies to point to desired system's package.

Now download the dependencies and build a firmware archive:
  $ cd dht_test
  $ mix deps.get
  $ mix firmware

If your target boots up using an SDCard (like the Raspberry Pi 3),
then insert an SDCard into a reader on your computer and run:
  $ mix firmware.burn

Plug the SDCard into the target and power it up. See target documentation
above for more information and other targets.

sevus:p craigbowes$ cd dht_test/
sevus:dht_test craigbowes$ vi mix.exs
sevus:dht_test craigbowes$ mix deps.get
sevus:dht_test craigbowes$ echo MIX_TARGET
MIX_TARGET
sevus:dht_test craigbowes$ echo $MIX_TARGET
rpi2
sevus:dht_test craigbowes$ vi mix.exs 
```
Here i added the dependency on the library....


```
sevus:dht_test craigbowes$ mix deps.get
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  dns 2.1.2
  elixir_make 0.6.0
  mdns 1.0.3
  muontrap 0.5.0
  nerves 1.5.1
  nerves_firmware_ssh 0.4.4
  nerves_init_gadget 0.7.0
  nerves_network 0.5.5
  nerves_network_interface 0.4.6
  nerves_runtime 0.10.3
  nerves_system_bbb 2.4.0
  nerves_system_br 1.9.2
  nerves_system_linter 0.3.0
  nerves_system_rpi 1.9.0
  nerves_system_rpi0 1.9.0
  nerves_system_rpi2 1.9.0
  nerves_system_rpi3 1.9.0
  nerves_system_rpi3a 1.9.0
  nerves_system_rpi4 1.9.0
  nerves_system_x86_64 1.9.0
  nerves_time 0.3.1
  nerves_toolchain_arm_unknown_linux_gnueabihf 1.2.0
  nerves_toolchain_armv6_rpi_linux_gnueabi 1.2.0
  nerves_toolchain_ctng 1.6.0
  nerves_toolchain_x86_64_unknown_linux_musl 1.2.0
  nerves_wpa_supplicant 0.5.2
  one_dhcpd 0.2.3
  ring_logger 0.8.0
  shoehorn 0.6.0
  socket 0.3.13
  system_registry 0.8.2
  toolshed 0.2.11
  uboot_env 0.1.1
New:
  nerves_dht 0.1.1
* Getting nerves_dht (Hex package)

Nerves environment
  MIX_TARGET:   rpi2
  MIX_ENV:      dev

==> nerves
Compiling 37 files (.ex)
Generated nerves app
==> dht_test
Resolving Nerves artifacts...
  Cached nerves_system_rpi2
  Cached nerves_toolchain_arm_unknown_linux_gnueabihf
sevus:dht_test craigbowes$ mix firmware

Nerves environment
  MIX_TARGET:   rpi2
  MIX_ENV:      dev

==> nerves_system_br
Generated nerves_system_br app
==> socket
Compiling 11 files (.ex)
Generated socket app
==> uboot_env
Compiling 7 files (.ex)
Generated uboot_env app
==> nerves_toolchain_ctng
Compiling 1 file (.ex)
Generated nerves_toolchain_ctng app
==> ring_logger
Compiling 4 files (.ex)
Generated ring_logger app
==> nerves_system_linter
Compiling 14 files (.ex)
Generated nerves_system_linter app
==> dns
Compiling 7 files (.ex)
Generated dns app
==> mdns
Compiling 5 files (.ex)
Generated mdns app
==> nerves_toolchain_arm_unknown_linux_gnueabihf
Generated nerves_toolchain_arm_unknown_linux_gnueabihf app
==> system_registry
Compiling 13 files (.ex)
Generated system_registry app
==> shoehorn
Compiling 8 files (.ex)
Generated shoehorn app
==> elixir_make
Compiling 1 file (.ex)
Generated elixir_make app
==> nerves_runtime
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/nerves_runtime/obj
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/nerves_runtime/priv
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -std=gnu99 -o /projects/dht_test/_build/rpi2_dev/lib/nerves_runtime/obj/nerves_runtime.o src/nerves_runtime.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -std=gnu99 -o /projects/dht_test/_build/rpi2_dev/lib/nerves_runtime/obj/uevent.o src/uevent.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -std=gnu99 -o /projects/dht_test/_build/rpi2_dev/lib/nerves_runtime/obj/kmsg_tailer.o src/kmsg_tailer.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc /projects/dht_test/_build/rpi2_dev/lib/nerves_runtime/obj/nerves_runtime.o /projects/dht_test/_build/rpi2_dev/lib/nerves_runtime/obj/uevent.o /projects/dht_test/_build/rpi2_dev/lib/nerves_runtime/obj/kmsg_tailer.o -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/lib -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/lib -lerts -lerl_interface -lei --sysroot=/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging -lmnl -o /projects/dht_test/_build/rpi2_dev/lib/nerves_runtime/priv/nerves_runtime
Compiling 13 files (.ex)
Generated nerves_runtime app
==> toolshed
Compiling 9 files (.ex)
Generated toolshed app
==> nerves_firmware_ssh
Compiling 9 files (.ex)
Generated nerves_firmware_ssh app
==> nerves_dht
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -o src/common_dht_read.o src/common_dht_read.c -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -o src/Raspberry_Pi_2/pi_2_dht_read.o src/Raspberry_Pi_2/pi_2_dht_read.c -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -o src/Raspberry_Pi_2/pi_2_mmio.o src/Raspberry_Pi_2/pi_2_mmio.c -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -o src/_Raspberry_Pi_2_Driver.o src/_Raspberry_Pi_2_Driver.c -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include
src/_Raspberry_Pi_2_Driver.c: In function 'main':
src/_Raspberry_Pi_2_Driver.c:26:3: warning: 'erl_init' is deprecated [-Wdeprecated-declarations]
   erl_init(NULL, 0);
   ^~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:325:8: note: declared here
 void   erl_init(void *x, long y) EI_DEPRECATED_ATTR;
        ^~~~~~~~
src/_Raspberry_Pi_2_Driver.c:35:7: warning: 'erl_mk_atom' is deprecated [-Wdeprecated-declarations]
       arr[0] = erl_mk_atom("ok");
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:383:8: note: declared here
 ETERM *erl_mk_atom(const char*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:36:7: warning: 'erl_mk_int' is deprecated [-Wdeprecated-declarations]
       arr[1] = erl_mk_int(pin);
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:388:8: note: declared here
 ETERM *erl_mk_int(int) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:37:7: warning: 'erl_mk_int' is deprecated [-Wdeprecated-declarations]
       arr[2] = erl_mk_int(sensor);
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:388:8: note: declared here
 ETERM *erl_mk_int(int) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:38:7: warning: 'erl_mk_float' is deprecated [-Wdeprecated-declarations]
       arr[3] = erl_mk_float(humidity);
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:387:8: note: declared here
 ETERM *erl_mk_float(double) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:39:7: warning: 'erl_mk_float' is deprecated [-Wdeprecated-declarations]
       arr[4] = erl_mk_float(temperature);
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:387:8: note: declared here
 ETERM *erl_mk_float(double) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:41:7: warning: 'erl_mk_tuple' is deprecated [-Wdeprecated-declarations]
       tuple  = erl_mk_tuple(arr, 5);
       ^~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:397:8: note: declared here
 ETERM *erl_mk_tuple(ETERM**,int) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:43:7: warning: 'erl_encode' is deprecated [-Wdeprecated-declarations]
       erl_encode(tuple, buf);
       ^~~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:428:8: note: declared here
 int    erl_encode(ETERM*,unsigned char*t) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:45:7: warning: 'erl_term_len' is deprecated [-Wdeprecated-declarations]
       write_cmd(buf, erl_term_len(tuple));
       ^~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:433:8: note: declared here
 int    erl_term_len(ETERM*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:47:7: warning: 'erl_free_term' is deprecated [-Wdeprecated-declarations]
       erl_free_term(tuple);
       ^~~~~~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:420:8: note: declared here
 void   erl_free_term(ETERM*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:49:7: warning: 'erl_mk_atom' is deprecated [-Wdeprecated-declarations]
       arr[0] = erl_mk_atom("error");
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:383:8: note: declared here
 ETERM *erl_mk_atom(const char*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:50:7: warning: 'erl_mk_int' is deprecated [-Wdeprecated-declarations]
       arr[1] = erl_mk_int(result);
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:388:8: note: declared here
 ETERM *erl_mk_int(int) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:52:7: warning: 'erl_mk_tuple' is deprecated [-Wdeprecated-declarations]
       tuple  = erl_mk_tuple(arr, 2);
       ^~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:397:8: note: declared here
 ETERM *erl_mk_tuple(ETERM**,int) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:54:7: warning: 'erl_encode' is deprecated [-Wdeprecated-declarations]
       erl_encode(tuple, buf);
       ^~~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:428:8: note: declared here
 int    erl_encode(ETERM*,unsigned char*t) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:56:7: warning: 'erl_term_len' is deprecated [-Wdeprecated-declarations]
       write_cmd(buf, erl_term_len(tuple));
       ^~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:433:8: note: declared here
 int    erl_term_len(ETERM*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:58:7: warning: 'erl_free_term' is deprecated [-Wdeprecated-declarations]
       erl_free_term(tuple);
       ^~~~~~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:420:8: note: declared here
 void   erl_free_term(ETERM*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~~
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc src/common_dht_read.o src/Raspberry_Pi_2/pi_2_dht_read.o src/Raspberry_Pi_2/pi_2_mmio.o src/_Raspberry_Pi_2_Driver.o -o priv/nerves_dht -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/lib -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/lib -lerts -lerl_interface -lei --sysroot=/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging  -lpthread
Compiling 1 file (.ex)
Generated nerves_dht app
==> one_dhcpd
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/one_dhcpd/priv
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/one_dhcpd/obj
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -o /projects/dht_test/_build/rpi2_dev/lib/one_dhcpd/obj/arp_set.o src/arp_set.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -o /projects/dht_test/_build/rpi2_dev/lib/one_dhcpd/priv/arp_set --sysroot=/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging /projects/dht_test/_build/rpi2_dev/lib/one_dhcpd/obj/arp_set.o
Compiling 9 files (.ex)
Generated one_dhcpd app
==> muontrap
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src all
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/muontrap/priv
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/muontrap/obj
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -std=c99 -D_GNU_SOURCE -o /projects/dht_test/_build/rpi2_dev/lib/muontrap/obj/muontrap.o muontrap.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc /projects/dht_test/_build/rpi2_dev/lib/muontrap/obj/muontrap.o --sysroot=/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging  -o /projects/dht_test/_build/rpi2_dev/lib/muontrap/priv/muontrap
if [ -f test/Makefile ]; then /Applications/Xcode.app/Contents/Developer/usr/bin/make -C test; fi
Compiling 5 files (.ex)
Generated muontrap app
==> nerves_network_interface
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/nerves_network_interface/ebin/../priv
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/nerves_network_interface/ebin/../obj
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -std=c99 -D_XOPEN_SOURCE=600 -o /projects/dht_test/_build/rpi2_dev/lib/nerves_network_interface/ebin/../obj/erlcmd.o src/erlcmd.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -std=c99 -D_XOPEN_SOURCE=600 -o /projects/dht_test/_build/rpi2_dev/lib/nerves_network_interface/ebin/../obj/netif.o src/netif.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc /projects/dht_test/_build/rpi2_dev/lib/nerves_network_interface/ebin/../obj/erlcmd.o /projects/dht_test/_build/rpi2_dev/lib/nerves_network_interface/ebin/../obj/netif.o -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/lib -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/lib -lerts -lerl_interface -lei --sysroot=/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging -lmnl -o /projects/dht_test/_build/rpi2_dev/lib/nerves_network_interface/ebin/../priv/netif
# setuid root net_basic so that it can configure network interfaces
SUDO_ASKPASS= true -- sh -c 'chown root:root /projects/dht_test/_build/rpi2_dev/lib/nerves_network_interface/ebin/../priv/netif; chmod +s /projects/dht_test/_build/rpi2_dev/lib/nerves_network_interface/ebin/../priv/netif'
Compiling 3 files (.ex)
Generated nerves_network_interface app
==> nerves_wpa_supplicant
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../priv
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../obj/wpa_ctrl
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -DCONFIG_CTRL_IFACE -DCONFIG_CTRL_IFACE_UNIX -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -o /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../obj/wpa_ex.o src/wpa_ex.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -DCONFIG_CTRL_IFACE -DCONFIG_CTRL_IFACE_UNIX -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -o /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../obj/wpa_ctrl/os_unix.o src/wpa_ctrl/os_unix.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -DCONFIG_CTRL_IFACE -DCONFIG_CTRL_IFACE_UNIX -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -o /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../obj/wpa_ctrl/wpa_ctrl.o src/wpa_ctrl/wpa_ctrl.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../obj/wpa_ex.o /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../obj/wpa_ctrl/os_unix.o /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../obj/wpa_ctrl/wpa_ctrl.o --sysroot=/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging -lrt -o /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../priv/wpa_ex
# setuid root wpa_ex so that it can interact with the wpa_supplicant
SUDO_ASKPASS= true -- sh -c 'chown root:root /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../priv/wpa_ex; chmod +s /projects/dht_test/_build/rpi2_dev/lib/nerves_wpa_supplicant/ebin/../priv/wpa_ex'
Compiling 3 files (.ex)
Generated nerves_wpa_supplicant app
==> nerves_network
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/nerves_network/ebin/../obj
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/nerves_network/ebin/../priv
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -std=c99 -o /projects/dht_test/_build/rpi2_dev/lib/nerves_network/ebin/../obj/udhcpc_wrapper.o src/udhcpc_wrapper.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc /projects/dht_test/_build/rpi2_dev/lib/nerves_network/ebin/../obj/udhcpc_wrapper.o -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/lib -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/lib -lerts -lerl_interface -lei --sysroot=/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging  -o /projects/dht_test/_build/rpi2_dev/lib/nerves_network/ebin/../priv/udhcpc_wrapper
# setuid root udhcpc_wrapper so that it can call udhcpc
SUDO_ASKPASS= true -- sh -c 'chown root:root /projects/dht_test/_build/rpi2_dev/lib/nerves_network/ebin/../priv/udhcpc_wrapper; chmod +s /projects/dht_test/_build/rpi2_dev/lib/nerves_network/ebin/../priv/udhcpc_wrapper'
Compiling 12 files (.ex)
Generated nerves_network app
==> nerves_system_rpi2
Generated nerves_system_rpi2 app
==> nerves_time
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/nerves_time/priv
mkdir -p /projects/dht_test/_build/rpi2_dev/lib/nerves_time/obj
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/include -std=c99 -o /projects/dht_test/_build/rpi2_dev/lib/nerves_time/obj/ntpd_script.o src/ntpd_script.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc /projects/dht_test/_build/rpi2_dev/lib/nerves_time/obj/ntpd_script.o -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging/usr/lib/erlang/lib/erl_interface-3.12/lib -lei_st --sysroot=/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.9.0/staging  -o /projects/dht_test/_build/rpi2_dev/lib/nerves_time/priv/ntpd_script
Compiling 5 files (.ex)
Generated nerves_time app
==> nerves_init_gadget
Compiling 5 files (.ex)
Generated nerves_init_gadget app
==> dht_test
Compiling 2 files (.ex)
Generated dht_test app
|nerves_bootstrap| Building OTP Release...

* skipping runtime configuration (config/releases.exs not found)
* creating _build/rpi2_dev/rel/dht_test/releases/0.1.0/vm.args
Updating base firmware image with Erlang release...
scrub-otp-release.sh: ERROR: Unexpected executable format for '/p/dht_test/_build/_nerves-tmp/rootfs-additions/srv/erlang/lib/nerves_dht-0.1.1/priv/nerves_dht'

Got:
 readelf:Advanced Micro Devices X86-64;0x0

Expecting:
 readelf:ARM;0x5000400, Version5 EABI, hard-float ABI

This file was compiled for the host or a different target and probably
will not work.

Check the following:

1. Are you using a path dependency in your mix deps? If so, run
   'mix clean' in that directory to avoid pulling in any of its
   build products.

2. Did you recently upgrade to Elixir 1.9 or Nerves 1.5?
   Nerves 1.5 adds support for Elixir 1.9 Releases and requires
   you to either add an Elixir 1.9 Release configuration or add
   Distillery as a dependency. Without this, the OTP binaries
   for your build machine will get included incorrectly and cause
   this error. See
   https://hexdocs.pm/nerves/updating-projects.html#updating-from-v1-4-to-v1-5

3. Did you recently upgrade or change your Nerves system? If so,
   try cleaning and rebuilding this project and its deps.

4. Are you building outside of Nerves' mix integration? If so,
   make sure that you've sourced 'nerves-env.sh'.

If you're still having trouble, please file an issue on Github
at https://github.com/nerves-project/nerves_system_br/issues.

** (Mix) Nerves encountered an error. %IO.Stream{device: :standard_io, line_or_bytes: :line, raw: true}
```

I had a similar error on an existing project where I was trying to rebuild using your release instead of jjcarsten's fork.  I got this:

```
 mix deps.get && mix firmware
Resolving Hex dependencies...
Dependency resolution completed:
Unchanged:
  dns 2.1.2
  elixir_make 0.6.0
  mdns 1.0.3
  muontrap 0.4.4
  nerves 1.5.1
  nerves_dht 0.1.1
  nerves_firmware_ssh 0.4.4
  nerves_init_gadget 0.7.0
  nerves_network 0.5.5
  nerves_network_interface 0.4.6
  nerves_runtime 0.10.2
  nerves_system_bbb 2.3.2
  nerves_system_br 1.8.5
  nerves_system_linter 0.3.0
  nerves_system_rpi 1.8.2
  nerves_system_rpi0 1.8.2
  nerves_system_rpi2 1.8.2
  nerves_system_rpi3 1.8.2
  nerves_system_rpi3a 1.8.2
  nerves_system_rpi4 1.8.2
  nerves_system_x86_64 1.8.2
  nerves_time 0.2.1
  nerves_toolchain_arm_unknown_linux_gnueabihf 1.2.0
  nerves_toolchain_armv6_rpi_linux_gnueabi 1.2.0
  nerves_toolchain_ctng 1.6.0
  nerves_toolchain_x86_64_unknown_linux_musl 1.2.0
  nerves_wpa_supplicant 0.5.2
  one_dhcpd 0.2.2
  ring_logger 0.8.0
  shoehorn 0.6.0
  socket 0.3.13
  system_registry 0.8.2
  toolshed 0.2.10
  uboot_env 0.1.1
* Getting nerves (Hex package)
* Getting shoehorn (Hex package)
* Getting ring_logger (Hex package)
* Getting toolshed (Hex package)
* Getting nerves_runtime (Hex package)
* Getting nerves_init_gadget (Hex package)
* Getting nerves_system_rpi (Hex package)
* Getting nerves_system_rpi0 (Hex package)
* Getting nerves_system_rpi2 (Hex package)
* Getting nerves_system_rpi3 (Hex package)
* Getting nerves_system_rpi3a (Hex package)
* Getting nerves_system_rpi4 (Hex package)
* Getting nerves_system_bbb (Hex package)
* Getting nerves_system_x86_64 (Hex package)
* Getting nerves_dht (Hex package)
* Getting nerves_network (Hex package)
* Getting elixir_make (Hex package)
* Getting nerves_network_interface (Hex package)
* Getting nerves_wpa_supplicant (Hex package)
* Getting one_dhcpd (Hex package)
* Getting system_registry (Hex package)
* Getting nerves_system_br (Hex package)
* Getting nerves_system_linter (Hex package)
* Getting nerves_toolchain_x86_64_unknown_linux_musl (Hex package)
* Getting nerves_toolchain_ctng (Hex package)
* Getting nerves_toolchain_arm_unknown_linux_gnueabihf (Hex package)
* Getting nerves_toolchain_armv6_rpi_linux_gnueabi (Hex package)
* Getting mdns (Hex package)
* Getting nerves_firmware_ssh (Hex package)
* Getting nerves_time (Hex package)
* Getting muontrap (Hex package)
* Getting dns (Hex package)
* Getting socket (Hex package)
* Getting uboot_env (Hex package)

Nerves environment
  MIX_TARGET:   rpi2
  MIX_ENV:      dev

==> nerves
Compiling 37 files (.ex)
Generated nerves app
==> nerves_thermo
Resolving Nerves artifacts...
  Cached nerves_system_rpi2
  Cached nerves_toolchain_arm_unknown_linux_gnueabihf

Nerves environment
  MIX_TARGET:   rpi2
  MIX_ENV:      dev

==> nerves_system_br
Generated nerves_system_br app
==> socket
Compiling 11 files (.ex)
Generated socket app
==> uboot_env
Compiling 7 files (.ex)
Generated uboot_env app
==> nerves_toolchain_ctng
Compiling 1 file (.ex)
Generated nerves_toolchain_ctng app
==> ring_logger
Compiling 4 files (.ex)
Generated ring_logger app
==> nerves_system_linter
Compiling 14 files (.ex)
Generated nerves_system_linter app
==> dns
Compiling 7 files (.ex)
Generated dns app
==> mdns
Compiling 5 files (.ex)
Generated mdns app
==> nerves_toolchain_arm_unknown_linux_gnueabihf
Generated nerves_toolchain_arm_unknown_linux_gnueabihf app
==> system_registry
Compiling 13 files (.ex)
Generated system_registry app
==> shoehorn
Compiling 8 files (.ex)
Generated shoehorn app
==> elixir_make
Compiling 1 file (.ex)
Generated elixir_make app
==> nerves_runtime
mkdir -p /projects/nerves_thermo/_build/rpi2_dev/lib/nerves_runtime/ebin/../obj
mkdir -p /projects/nerves_thermo/_build/rpi2_dev/lib/nerves_runtime/ebin/../priv
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/include -std=gnu99 -o /projects/nerves_thermo/_build/rpi2_dev/lib/nerves_runtime/ebin/../obj/nerves_runtime.o src/nerves_runtime.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/include -std=gnu99 -o /projects/nerves_thermo/_build/rpi2_dev/lib/nerves_runtime/ebin/../obj/uevent.o src/uevent.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/include -std=gnu99 -o /projects/nerves_thermo/_build/rpi2_dev/lib/nerves_runtime/ebin/../obj/kmsg_tailer.o src/kmsg_tailer.c
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc /projects/nerves_thermo/_build/rpi2_dev/lib/nerves_runtime/ebin/../obj/nerves_runtime.o /projects/nerves_thermo/_build/rpi2_dev/lib/nerves_runtime/ebin/../obj/uevent.o /projects/nerves_thermo/_build/rpi2_dev/lib/nerves_runtime/ebin/../obj/kmsg_tailer.o -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/erts-10.4.4/lib -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/lib -lerts -lerl_interface -lei --sysroot=/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging -lmnl -o /projects/nerves_thermo/_build/rpi2_dev/lib/nerves_runtime/ebin/../priv/nerves_runtime
Compiling 13 files (.ex)
Generated nerves_runtime app
==> toolshed
Compiling 9 files (.ex)
Generated toolshed app
==> nerves_firmware_ssh
Compiling 9 files (.ex)
Generated nerves_firmware_ssh app
==> nerves_dht
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -o src/Raspberry_Pi_2/pi_2_dht_read.o src/Raspberry_Pi_2/pi_2_dht_read.c -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/include
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -o src/Raspberry_Pi_2/pi_2_mmio.o src/Raspberry_Pi_2/pi_2_mmio.c -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/include
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc -o src/_Raspberry_Pi_2_Driver.o src/_Raspberry_Pi_2_Driver.c -c -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/erts-10.4.4/include -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  -pipe -Os -I/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/include
src/_Raspberry_Pi_2_Driver.c: In function 'main':
src/_Raspberry_Pi_2_Driver.c:26:3: warning: 'erl_init' is deprecated [-Wdeprecated-declarations]
   erl_init(NULL, 0);
   ^~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:325:8: note: declared here
 void   erl_init(void *x, long y) EI_DEPRECATED_ATTR;
        ^~~~~~~~
src/_Raspberry_Pi_2_Driver.c:35:7: warning: 'erl_mk_atom' is deprecated [-Wdeprecated-declarations]
       arr[0] = erl_mk_atom("ok");
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:383:8: note: declared here
 ETERM *erl_mk_atom(const char*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:36:7: warning: 'erl_mk_int' is deprecated [-Wdeprecated-declarations]
       arr[1] = erl_mk_int(pin);
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:388:8: note: declared here
 ETERM *erl_mk_int(int) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:37:7: warning: 'erl_mk_int' is deprecated [-Wdeprecated-declarations]
       arr[2] = erl_mk_int(sensor);
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:388:8: note: declared here
 ETERM *erl_mk_int(int) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:38:7: warning: 'erl_mk_float' is deprecated [-Wdeprecated-declarations]
       arr[3] = erl_mk_float(humidity);
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:387:8: note: declared here
 ETERM *erl_mk_float(double) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:39:7: warning: 'erl_mk_float' is deprecated [-Wdeprecated-declarations]
       arr[4] = erl_mk_float(temperature);
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:387:8: note: declared here
 ETERM *erl_mk_float(double) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:41:7: warning: 'erl_mk_tuple' is deprecated [-Wdeprecated-declarations]
       tuple  = erl_mk_tuple(arr, 5);
       ^~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:397:8: note: declared here
 ETERM *erl_mk_tuple(ETERM**,int) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:43:7: warning: 'erl_encode' is deprecated [-Wdeprecated-declarations]
       erl_encode(tuple, buf);
       ^~~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:428:8: note: declared here
 int    erl_encode(ETERM*,unsigned char*t) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:45:7: warning: 'erl_term_len' is deprecated [-Wdeprecated-declarations]
       write_cmd(buf, erl_term_len(tuple));
       ^~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:433:8: note: declared here
 int    erl_term_len(ETERM*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:47:7: warning: 'erl_free_term' is deprecated [-Wdeprecated-declarations]
       erl_free_term(tuple);
       ^~~~~~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:420:8: note: declared here
 void   erl_free_term(ETERM*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:49:7: warning: 'erl_mk_atom' is deprecated [-Wdeprecated-declarations]
       arr[0] = erl_mk_atom("error");
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:383:8: note: declared here
 ETERM *erl_mk_atom(const char*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:50:7: warning: 'erl_mk_int' is deprecated [-Wdeprecated-declarations]
       arr[1] = erl_mk_int(result);
       ^~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:388:8: note: declared here
 ETERM *erl_mk_int(int) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:52:7: warning: 'erl_mk_tuple' is deprecated [-Wdeprecated-declarations]
       tuple  = erl_mk_tuple(arr, 2);
       ^~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:397:8: note: declared here
 ETERM *erl_mk_tuple(ETERM**,int) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:54:7: warning: 'erl_encode' is deprecated [-Wdeprecated-declarations]
       erl_encode(tuple, buf);
       ^~~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:428:8: note: declared here
 int    erl_encode(ETERM*,unsigned char*t) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:56:7: warning: 'erl_term_len' is deprecated [-Wdeprecated-declarations]
       write_cmd(buf, erl_term_len(tuple));
       ^~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:433:8: note: declared here
 int    erl_term_len(ETERM*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~
src/_Raspberry_Pi_2_Driver.c:58:7: warning: 'erl_free_term' is deprecated [-Wdeprecated-declarations]
       erl_free_term(tuple);
       ^~~~~~~~~~~~~
In file included from src/_Raspberry_Pi_2_Driver.c:6:
/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/include/erl_interface.h:420:8: note: declared here
 void   erl_free_term(ETERM*) EI_DEPRECATED_ATTR;
        ^~~~~~~~~~~~~
/Users/craigbowes/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-darwin_x86_64-1.2.0/bin/arm-unknown-linux-gnueabihf-gcc src/common_dht_read.o src/Raspberry_Pi_2/pi_2_dht_read.o src/Raspberry_Pi_2/pi_2_mmio.o src/_Raspberry_Pi_2_Driver.o -o priv/nerves_dht -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/erts-10.4.4/lib -L/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging/usr/lib/erlang/lib/erl_interface-3.12/lib -lerts -lerl_interface -lei --sysroot=/Users/craigbowes/.nerves/artifacts/nerves_system_rpi2-portable-1.8.2/staging  -lpthread
src/common_dht_read.o: file not recognized: file format not recognized
collect2: error: ld returned 1 exit status
make: *** [priv/nerves_dht] Error 1
could not compile dependency :nerves_dht, "mix compile" failed. You can recompile this dependency with "mix deps.compile nerves_dht", update it with "mix deps.update nerves_dht" or clean it with "mix deps.clean nerves_dht"
 22
==> nerves_thermo
** (Mix) Could not compile with "make" (exit status: 2).
You need to have gcc and make installed. Try running the
commands "gcc --version" and / or "make --version". If these programs
are not installed, you will be prompted to install them.
```

let me know if you have any questions.    one of the core nerves guys told me this on slack FWIW:

>>>It looks like nerves_dht leaves its output in the main source directory (not under _build). This has started causing a lot of issues and we’ve been fixing projects over the past year to avoid this. Building clean is somewhat of a workaround

--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,8 @@
+TOP := $(dir $(realpath $(lastword $(MAKEFILE_LIST))))
+
+PREFIX = $(MIX_COMPILE_PATH)/../priv
+BUILD  = $(MIX_COMPILE_PATH)/../obj
+
 # Look for the EI library and header files
 # For crosscompiled builds, ERL_EI_INCLUDE_DIR and ERL_EI_LIBDIR must be
 # passed into the Makefile.
@@ -14,59 +19,43 @@ endif
 ERL_CFLAGS ?= -I$(ERL_EI_INCLUDE_DIR)
 ERL_LDFLAGS ?= -L$(ERL_EI_LIBDIR) -lerl_interface -lei
 
-SRC = src/common_dht_read.c
 LDFLAGS +=
 CFLAGS ?= -O2 -Wall -Wextra -Wno-unused-parameter
 
-ifeq ($(MIX_TARGET), rpi)
-	SRC += $(wildcard src/Raspberry_Pi/*.c) src/_Raspberry_Pi_Driver.c
+RPI := rpi rpi0
+RPI2 := rpi2 rpi3 rpi3a rpi4
+BBB := bbb
+
+ifneq ($(filter $(MIX_TARGET),$(RPI)),)
+	SRC += $(wildcard src/Raspberry_Pi/*.c) src/_Raspberry_Pi_Driver.c src/common_dht_read.c
+else ifneq ($(filter $(MIX_TARGET),$(RPI2)),)
+	SRC += $(wildcard src/Raspberry_Pi_2/*.c) src/_Raspberry_Pi_2_Driver.c src/common_dht_read.c
+else ifneq ($(filter $(MIX_TARGET),$(BBB)),)
+	SRC += $(wildcard src/Beaglebone_Black/*.c) src/_Beaglebone_Black_Driver.c src/common_dht_read.c
+else ifeq ($(MIX_TARGET), host)
+	SRC = src/_Host_Driver.c
+	CC = gcc
 endif
 
-ifeq ($(MIX_TARGET), rpi0)
-	SRC += $(wildcard src/Raspberry_Pi/*.c) src/_Raspberry_Pi_Driver.c
-endif
+CC ?= $(CROSSCOMPILER)-gcc
 
-ifeq ($(MIX_TARGET), rpi2)
-	SRC += $(wildcard src/Raspberry_Pi_2/*.c) src/_Raspberry_Pi_2_Driver.c
-endif
+OBJ = $(foreach file,$(SRC),$(BUILD)/$(notdir $(file:.c=.o)))
 
-ifeq ($(MIX_TARGET), rpi3)
-	SRC += $(wildcard src/Raspberry_Pi_2/*.c) src/_Raspberry_Pi_2_Driver.c
-endif
+calling_from_make:
+	mix compile
 
-ifeq ($(MIX_TARGET), rpi4)
-	SRC += $(wildcard src/Raspberry_Pi_2/*.c) src/_Raspberry_Pi_2_Driver.c
-endif
+all: $(PREFIX)/nerves_dht;
 
-ifeq ($(MIX_TARGET), rpi3a)
-	SRC += $(wildcard src/Raspberry_Pi_2/*.c) src/_Raspberry_Pi_2_Driver.c
-endif
+%.o:
+	$(CC) -c $(ERL_CFLAGS) $(CFLAGS) -o $@ $(filter %$(notdir $(basename $@)).c, $(SRC))
 
-ifeq ($(MIX_TARGET), bbb)
-	SRC += $(wildcard src/Beaglebone_Black/*.c) src/_Beaglebone_Black_Driver.c
-endif
+$(PREFIX) $(BUILD):
+	mkdir -p $@
 
-ifeq ($(MIX_TARGET), host)
-	SRC += src/_Host_Driver.c
-	CC ?= gcc
-else
-	CC ?= $(CROSSCOMPILER)-gcc
-endif
-
-OBJ = $(SRC:.c=.o)
-
-.PHONY: all clean
-
-all: priv priv/nerves_dht
-
-%.o: %.c
-	$(CC) -o $@ $^ -c $(ERL_CFLAGS) $(CFLAGS)
-
-priv:
-	mkdir -p priv
-
-priv/nerves_dht: $(OBJ)
-	$(CC) $^ -o $@ $(ERL_LDFLAGS) $(LDFLAGS) -lpthread
+$(PREFIX)/nerves_dht: $(BUILD) $(PREFIX) $(OBJ)
+	$(CC) $(wildcard $(BUILD)/*.o) -o $@ $(ERL_LDFLAGS) $(LDFLAGS) -lpthread
 
 clean:
-	rm -f priv/nerves_dht src/*.o src/**/*.o
+	if [ -n "$(MIX_COMPILE_PATH)" ]; then $(RM) -r $(BUILD); $(RM) -r $(PREFIX); fi
+
+.PHONY: all clean calling_from_make

--- a/mix.exs
+++ b/mix.exs
@@ -7,6 +7,8 @@ defmodule NervesDht.Mixfile do
      description: description(),
      package: package(),
      compilers: [:elixir_make] ++ Mix.compilers,
+     make_targets: ["all"],
+     make_clean: ["clean"],
      build_embedded: true,
      start_permanent: Mix.env == :prod,
      deps: deps()]

